### PR TITLE
rename directories and namespaces from ssl to tls

### PR DESCRIPTION
**Pull request by bdecoste**
Signed-off-by: William DeCoste <bdecoste@gmail.com>

For an explanation of how to fill out the fields, please see the relevant section 
in [PULL_REQUESTS.md](./PULL_REQUESTS.md)

*Description*: Rename all directories from ssl to tls and all namespaces from Ssl to Tls. Makes all of the directories and namespaces consistent. Use Envoy:: namespace prefix only when needed to avoid conflicts. 
*Risk Level*: Low
*Testing*: Standard tests passed
*Docs Changes*: None
*Release Notes*: None
[Optional Fixes #Issue]
[Optional *Deprecated*:]

### CODE REVIEW

File: include/envoy/network/transport_socket.h [hunk 1]
```diff
@@ -133,7 +133,7 @@ class TransportSocket {
   /**
    * @return the const SSL connection data if this is an SSL connection, or nullptr if it is not.
```
Comment (lizan) [inline include/envoy/network/transport_socket.h]: s/SSL/TLS/?

File: include/envoy/network/transport_socket.h [hunk 2]
```diff
@@ -133,7 +133,7 @@ class TransportSocket {
...
    */
-  virtual const Ssl::Connection* ssl() const PURE;
+  virtual const Tls::Connection* ssl() const PURE;
```
Comment (lizan) [inline include/envoy/network/transport_socket.h]: tls?

File: include/envoy/server/transport_socket_config.h [hunk 3]
```diff
@@ -26,9 +26,9 @@ class TransportSocketFactoryContext {
   virtual ~TransportSocketFactoryContext() {}
 
   /**
-   * @return Ssl::ContextManager& the SSL context manager.
+   * @return Tls::ContextManager& the SSL context manager.
```
Comment (lizan) [inline include/envoy/server/transport_socket_config.h]: TLS context manager?

File: include/envoy/tls/connection.h [hunk 4]
```diff
@@ -6,7 +6,7 @@
 #include "envoy/common/pure.h"
 
 namespace Envoy {
-namespace Ssl {
+namespace Tls {
 
 /**
  * Base connection interface for all SSL connections.
```
Comment (lizan) [inline include/envoy/tls/connection.h]: TLS

**Comment by ggreenway** [comment]
I agree with this rename, but slightly worry about confusion with thread-local storage. Can someone do a quick grep for "tls" in comments/names and find any that are referring to thread-local and see if there's any way someone would be confused in the context they appear?

**Comment by htuch** [comment]
Yeah, it would be nice to have two different terms for code search purposes, even though this is more technically correct.

File: include/envoy/network/transport_socket.h (hunk 1 above)
Comment (bdecoste) [reply include/envoy/network/transport_socket.h]: I will correct the comments asap. Thanks

File: include/envoy/tls/connection.h (hunk 4 above)
Comment (bdecoste) [reply include/envoy/tls/connection.h]: I'm not sure what you mean by these comments. I moved include/envoy/ssl and Envoy::Ssl to include/envoy/tls and Envoy::Tls::

Thanks

**Comment by bdecoste** [comment]
I will check for conflicts between tls and thread-local storage. Thanks.

**Comment by bdecoste** [comment]
There are references to TLS, Tls, and tls that refer to thread-local store.

**Comment by mattklein123** [comment]
@lizan @PiotrSikora @htuch @ggreenway thoughts here? I don't really care that much one way or the other, but to me, SSL vs. TLS (thread local storage) is easier to understand and everyone knows what it means.

**Comment by ggreenway** [comment]
I think we at least need to be consistent, although the situation is a mess. Over a decade ago, the protocol was renamed to tls, but all of the common libraries implementing it still have "ssl" in their name.

And in Envoy, we have `source/common/ssl`, and we also have `source/extensions/transport_sockets/tls/ssl_socket`. 

I would rather standardize on TLS, because that's the proper name of the protocol, but if we decide on SSL I think we should go through and cleanup all references so they're similar.

Except, we can't change the name of the TLS transport socket in configuration because that would break config. Which leads me further to thinking we should use TLS, and try to cleanup docs so that thread-local things spell it out as thread-local, not TLS.

**Comment by htuch** [comment]
I prefer SSL all things being equal, because it avoids overload, but I agree on the challenges of harmonizing. I guess I'd vote no on a massive rename that will create more conflation, but if enough other folks feel the other way I'm good with it.

**Comment by lizan** [comment]
I don't have strong opinion but prefer TLS slightly. I feel it is less confusing as long as they lived in extensions/tls and/or common/tls.

**Comment by mattklein123** [comment]
> I prefer SSL all things being equal, because it avoids overload, but I agree on the challenges of harmonizing. I guess I'd vote no on a massive rename that will create more conflation, but if enough other folks feel the other way I'm good with it.

I feel the same way, but like @htuch I don't feel strongly about it. One thing to keep in mind is that `thread_local` is a reserved word in C++11, which is why we end up using `tls` as a variable name all over the place.

**Comment by PiotrSikora** [comment]
For some background: SSL is/was Netscape's protocol, TLS is the IETF protocol, which evolved from it, and TLS 1.0 was standardized 20 years ago (!), so using SSL is, strictly speaking, incorrect. This is very similar to SPDY and HTTP/2, or Google's QUIC and HTTP/3, really.

The SSL name unfortunately stuck around, and in reality more people know about SSL than TLS, though, I would imagine that people working with web servers know both names equally well.

Right now, we're pretty consistently using TLS externally (in config directives and documentation), but internally it's a mix of SSL for core code and interfaces (in namespace and class names, therefore filenames and directories) and TLS for extensions (TLS Inspector or now also TLS transport socket).

As far as I understand, the only blocker for renaming core code and interfaces to TLS is the worry that TLS acronym is conflicting with Thread-Local Storage, which is going to confuse people, but we're not exposing this externally to the users, and we're using `ThreadLocal` for the namespace and in the class names, so this is really about a few comments and variables (and mental association for a few people).

I'd prefer if we standardized on using TLS for the security protocol everywhere, but I'm fine with leaving things as-is, if people are too worried about the confusion. I'm definitely against renaming things to SSL.

**Comment by bdecoste** [comment]
OK, sounds like I should switch extensions/transport_sockets/tls back to ssl. Thanks. I'll close this one and put in a new PR asap.

**Comment by lizan** [comment]
> OK, sounds like I should switch extensions/transport_sockets/tls back to ssl. Thanks. I'll close this one and put in a new PR asap.

IMO we should really keep the directory to "tls", all extensions directory matches their config name.

**Comment by bdecoste** [comment]
OK. Because the tls extensions gets included in the core source there will be tls search hits there. I'm going to submit the PR for https://github.com/envoyproxy/envoy/issues/5762 first. Thanks.

**Pull request closed without merging**